packages/mason_logger/lib/src/progress.dart

```diff
@@ -79,52 +86,77 @@ class Progress {
 
   int _index = 0;
 
+  String? _prevMessage;
+
   /// End the progress and mark it as completed.
   void complete([String? update]) {
     _stopwatch.stop();
     _write(
-      '''$_clearLine${lightGreen.wrap('✓')} ${update ?? _message} $_time\n''',
+      '''$_clearPrevMessage${lightGreen.wrap('✓')} ${update ?? _message} $_time\n''',
     );
     _timer?.cancel();
   }
 
   /// End the progress and mark it as failed.
   void fail([String? update]) {
     _timer?.cancel();
-    _write('$_clearLine${red.wrap('✗')} ${update ?? _message} $_time\n');
+    _write('$_clearPrevMessage${red.wrap('✗')} ${update ?? _message} $_time\n');
     _stopwatch.stop();
   }
 
   /// Update the progress message.
   void update(String update) {
-    if (_timer != null) _write(_clearLine);
+    if (_timer != null) _write(_clearPrevMessage);
     _message = update;
     _onTick(_timer);
   }
 
-  /// Cancel the progress and remove the written line.
+  /// Cancel the progress and remove the previous message.
   void cancel() {
     _timer?.cancel();
-    _write(_clearLine);
+    _write(_clearPrevMessage);
     _stopwatch.stop();
   }
 
-  String get _clearLine {
+  String get _clearPrevMessage {
```

Comment:
Can we add some tests for these changes? I would love to see some tests that fail without these changes and succeed with the changes. Also, can you describe in a bit more detail what issue you are resolving? If you could provide detailed reproduction steps for the specific issue that this fixes it would be super helpful, thanks!

Reply:
Yeah sure the problem is that in situations where the `message` written to `stdout` is longer than one line on the related terminal the the logger will repeatly print the `message` on every animation tick of `Logger.progress`. This is caused by `_clearLine` just deleting the last line (what works with single lines) but not multiline. A sample can be found in #711.

Reply:
Before:
![](https://raw.githubusercontent.com/jtdLab/gifs/main/before.gif)

After:
![](https://raw.githubusercontent.com/jtdLab/gifs/main/after.gif)

Reply:
@felangel There seems to be a problem with the test in `mason_logger_test.dart` line 519-551.
The test passes for some weird cases for example replace line 531-533 with 
```dart
stdout.write(
  '''${lightGreen.wrap('\b${'\b' * (message.length + 4 + time.length)}⠙')} SOME RANDOM TEXT... ${darkGray.wrap('(0.1s)')}''',
);
```

and the test still passes (might be a mocktail issue related to ANSI codes ?).